include/dlaf/matrix/internal/tile_future_manager.h

```diff
@@ -0,0 +1,85 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2019, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#include <hpx/hpx.hpp>
+#include "dlaf/matrix/tile.h"
+
+namespace dlaf {
+namespace matrix {
+namespace internal {
+
+template <class ReturnTileType, class TileType>
+hpx::future<ReturnTileType> setPromiseTileFuture(hpx::future<TileType>& old_future,
+                                                 hpx::lcos::local::promise<TileType>& p) noexcept {
```

Comment:
Just wanted to note that `future::then` is not `noexcept`, but if this is correctly set up the `noexcept` is of course warranted.